src/execution/execution.cc

```diff
@@ -343,6 +343,41 @@ V8_WARN_UNUSED_RESULT MaybeHandle<Object> Invoke(Isolate* isolate,
       DCHECK(!params.IsScript());
     }
 #endif
+
+    if (recordreplay::AreEventsDisallowed() &&
```

Comment:
This new block of code is quite complicated.  It would be better if most of this was factored into a helper function, then it will be more reusable and easier to see how this affects the Invoke() control flow.

Reply:
Refactored. Re-running tests now.